### PR TITLE
lightning: access aws resource by session token

### PR DESCRIPTION
**Pull request by okJiang**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/40750

Problem Summary:

Tools(BR/Dumpling/Lightning) do not support access AWS resources by AK/SK/token, but AK/SK only, which is not the best practice from AWS.

### What is changed and how it works?

add `session-token` to the related configuration.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)

1. make sure your role can access s3 resource
2. the `data-source-dir` example

```toml
[mydumper]
data-source-dir = "s3://xianjie-test-import-aksktoken?access_key=xxxxxxxxx&secret_access_key=xxxxxxxx&session-token=xxxxxxxx"
```

- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
BR/Dumpling/Lightning support access AWS resources by access-key/secret-access-key/session-token(optional)
```

### CODE REVIEW

**Comment by okJiang** [comment]
The `token` seems to not be testable in integration tests, but it has passed manual testing. ptal~ @dsdashun @lance6716

File: br/pkg/storage/parse_test.go [hunk 1]
```diff
@@ -69,14 +69,16 @@ func TestCreateStorage(t *testing.T) {
 	require.Equal(t, "TestKey", s3.SseKmsKeyId)
 
 	// special character in access keys
-	s, err = ParseBackend(`s3://bucket4/prefix/path?access-key=NXN7IPIOSAAKDEEOLMAF&secret-access-key=nREY/7Dt+PaIbYKrKlEEMMF/ExCiJEX=XMLPUANw`, nil)
+	// s, err = ParseBackend(`s3://bucket4/prefix/path?access-key=NXN7IPIOSAAKDEEOLMAF&secret-access-key=nREY/7Dt+PaIbYKrKlEEMMF/ExCiJEX=XMLPUANw`, nil)
```
Comment (lance6716) [inline br/pkg/storage/parse_test.go:72]: remove this comment?

**Review by lance6716 — APPROVED** [review]

**Pull request merged**